### PR TITLE
Add utility functions for creating mocked products

### PR DESCRIPTION
**Pull request by ardelato**
## Description

As a continuing effort to add tests for the Product Page, #452, we have added general utility functions for creating mocked products. These functions allow us to easily and quickly generate test data for a variety of product options that will be needed in addressing #452. 

## CR Notes:

Although these utility functions are not that exhaustive and pale in comparison to the `ProductExtension` class we have in our `iFixit/ifixit` repo, they should be enough to pick up some speed for creating the tests listed in #452.

## QA Notes:

I have tested that the mocked product data and utility functions work as expected:

<details>
<summary><strong>Discounted Products</strong></summary>

  https://user-images.githubusercontent.com/22064420/206541940-a885fbcb-17f6-46d6-8bce-3b271f56fc3c.mp4

https://user-images.githubusercontent.com/22064420/206541956-412c12f2-b315-4553-8e4f-fe5b778b5908.mp4

</details>

<details>
<summary><strong>Product by Type</strong></summary>

  

https://user-images.githubusercontent.com/22064420/206542182-fde0096f-e577-419c-873b-389734aa4710.mp4

</details>

<details>
<summary><strong>Product with Warranty</strong></summary>

https://user-images.githubusercontent.com/22064420/206542230-5a09b3fb-01e7-4fa0-ac86-9b46821a61ba.mp4

</details>

<br/> 

You can confirm the functions work by applying the following diff as well as checking that `CI` continues to work.

```diff
diff --git a/frontend/templates/product/sections/ProductSection/index.tsx b/frontend/templates/product/sections/ProductSection/index.tsx
index 45862b52..3e0d6594 100644
--- a/frontend/templates/product/sections/ProductSection/index.tsx
+++ b/frontend/templates/product/sections/ProductSection/index.tsx
@@ -43,6 +43,7 @@ import { ProductVideos } from './ProductVideos';
 import { Prop65Warning } from './Prop65Warning';
 import { useInternationalBuyBox } from '@templates/product/hooks/useInternationalBuyBox';
 import { InternationalBuyBox } from './InternationalBuyBox';
+import { getMockProduct } from 'test/jest/utils';
 
 export type ProductSectionProps = {
    product: Product;
@@ -70,6 +71,9 @@ export function ProductSection({
    );
    const isForSale = useIsProductForSale(product);
 
+   product = getMockProduct() // or other utility function to get a mock product
+   selectedVariant = product.variants[0]
+
    const compatibilityDrawerModelsTruncate = 4;
    const compatibilityDrawerDeviceTruncate = 3;
    const compatibilityDrawerIncomplete =
```

**cc:** @mlahargou 
Closes: #1076

### CODE REVIEW

File: frontend/test/jest/utils.tsx [hunk 1]
```diff
@@ -115,3 +118,14 @@ export const getNonDiscountedProduct = (originalPrice: number = 49.99) => {
    });
    return product;
 };
+
+export const getProductOfType = (type: 'battery' | 'tool' | 'part') => {
```
Comment (batbattur) [inline frontend/test/jest/utils.tsx:122]: Are we using this function anywhere or are we planning to replace it in a later pull?

Comment (batbattur) [reply frontend/test/jest/utils.tsx:122]: I was thinking of using this function in `ProductSection.test.tsx` but it looks like we don't think need to. 👍

**Review by batbattur — APPROVED** [review]
CR 🚛 💨 Awesome stuff @ardelato!

**Comment by deltuh-vee** [comment]
QA 🎬
Mocked product functions seem to be working and CI is still passing.

**Pull request merged**